src/foam/nanos/notification/NotificationExpansionDAO.js

```diff
@@ -0,0 +1,81 @@
+/**
+ * @license
+ * Copyright 2020 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+foam.CLASS({
+  package: 'foam.nanos.notification',
+  name: 'NotificationExpansionDAO',
+  extends: 'foam.dao.ProxyDAO',
+
+  documentation: `Expands a notification from a group or broadcast notification to a user specific notification. `,
+
+  implements: [
+    'foam.mlang.Expressions'
+  ],
+
+  imports: [
+    'notificationTemplateDAO'
+  ],
+
+  javaImports: [
+    'foam.core.Detachable',
+    'foam.core.FObject',
+    'foam.core.X',
+    'foam.dao.AbstractSink',
+    'foam.dao.DAO',
+    'foam.dao.ProxyDAO',
+    'foam.mlang.sink.Count',
+    'foam.nanos.auth.LifecycleState',
+    'foam.nanos.auth.User',
+    'foam.nanos.logger.Logger',
+    'foam.util.SafetyUtil',
+    'static foam.mlang.MLang.*'
+  ],
+
+  methods: [
+    {
+      name: 'put_',
+      javaCode: `
+        DAO userDAO = (DAO) x.get("localUserDAO");
+        Notification notif = (Notification) obj;
+    
+        if ( notif.getBroadcasted() ) {
+          userDAO.select(new AbstractSink() {
+            @Override
+            public void put(Object o, Detachable d) {
+              User user = (User) o;
+              user.doNotify(x, notif);
+            }
+          });
+        } else if ( ! SafetyUtil.isEmpty(notif.getGroupId()) ) {
+          DAO receivers = userDAO.where(
```

Comment:
This is an example where group should implement doNotify, then it can take care of notifying it's users.

Reply:
Could do in the future .. i'll create a follow-up FOAM ticket for it.

Reply:
https://nanopay.atlassian.net/browse/NP-1176 created.